lib/Runtime/Debug/TTSerialize.cpp

```diff
@@ -963,8 +966,8 @@ namespace TTD
             return NSTokens::ParseTokenKind::Error;
         }
 
-        //convert this number to get the length of the string (not including "")
-        charList.Add(_u('\0'));
+        // Convert this number to get the length of the string (not including ""),
+        // charList is already null-terminated by the call to ScanNumber.
```

Comment:
One thought I had was to add an assert to each location where we get the buffer in this code.  

Something like:

`TTDAssert(charList.Item(charList.Count() - 1) == '\0', "Char list must be null terminated");`